controllers/datadogagent/feature/cws/feature_test.go

```diff
@@ -0,0 +1,352 @@
+		sysProbeVolumeMounts := mgr.VolumeMountMgr.VolumeMountsByC[apicommonv1.SystemProbeContainerName]
+		assert.True(t, apiutils.IsEqualStruct(sysProbeVolumeMounts, sysprobeWantVolumeMount), "System probe volume mounts \ndiff = %s", cmp.Diff(sysProbeVolumeMounts, sysprobeWantVolumeMount))
+
+		// check volumes
+		wantVolumes := []corev1.Volume{
+			{
+				Name: apicommon.DebugfsVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: apicommon.DebugfsPath,
+					},
+				},
+			},
+			{
+				Name: apicommon.SecurityfsVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: apicommon.SecurityfsVolumePath,
+					},
+				},
+			},
+			{
+				Name: apicommon.SystemProbeSocketVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			},
+			{
+				Name: apicommon.ProcdirVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: apicommon.ProcdirHostPath,
+					},
+				},
+			},
+			{
+				Name: apicommon.PasswdVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: apicommon.PasswdHostPath,
+					},
+				},
+			},
+			{
+				Name: apicommon.GroupVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: apicommon.GroupHostPath,
+					},
+				},
+			},
+			{
+				Name: apicommon.SystemProbeOSReleaseDirVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: apicommon.SystemProbeOSReleaseDirVolumePath,
+					},
+				},
+			},
+			{
+				Name: apicommon.HostRootVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: apicommon.HostRootHostPath,
+					},
+				},
+			},
+			{
+				Name: apicommon.SecurityAgentRuntimeCustomPoliciesVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: "custom_test",
+						},
+					},
+				},
+			},
+			{
+				Name: apicommon.SecurityAgentRuntimePoliciesDirVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			},
+		}
+
+		volumes := mgr.VolumeMgr.Volumes
+		assert.True(t, apiutils.IsEqualStruct(volumes, wantVolumes), "Volumes \ndiff = %s", cmp.Diff(volumes, wantVolumes))
+	}
+
+	tests := test.FeatureTestSuite{
+		//////////////////////////
+		// v1Alpha1.DatadogAgent
+		//////////////////////////
+		{
+			Name:          "v1alpha1 CWS not enabled",
+			DDAv1:         ddav1CWSDisabled.DeepCopy(),
+			WantConfigure: false,
+		},
+		{
+			Name:          "v1alpha1 CWS enabled",
+			DDAv1:         ddav1CWSEnabled,
+			WantConfigure: true,
+			Agent: &test.ComponentTest{
```

Comment:
Nit: you can use the new helpers that we merged some days ago to avoid duplicating some code: https://github.com/DataDog/datadog-operator/blob/d0669c0682b56d716054ab45010de48b490bb925/controllers/datadogagent/feature/test/testsuite.go#L65

With that, you can remove the `createEmptyFakeManager` function defined at the top of this file, and here use: `Agent: test.NewDefaultComponentTest().WithWantFunc(cwsAgentNodeWantFunc)`. Same for the test below.